tests/serverpod_test_server/test_integration/auth_core/auth_header_test.dart

```diff
@@ -73,5 +73,61 @@ void main() async {
       var receivedToken = await tokenInspectionCompleter.future;
       expect(receivedToken, key);
     });
+
+    test(
+        'then old clients passing keys in body should still work with new server code',
+        () async {
+      var key = 'username-4713:password-4713';
+
+      // Intentionally clearing the authKeyManager so that auth is not passed in header.
+      await authKeyManager.remove();
+
+      // Verify that by calling it this way, the key is not passed in the header.
+      // This validates this unit test's approach.
+      tokenInspectionCompleter = Completer();
+      var reflectedHeader =
+          await client.reflection.caller.callServerEndpoint<List<String>?>(
+        'reflection',
+        'reflectHttpHeader',
+        {
+          'headerName': 'authorization',
+          'auth': key,
+        },
+      );
+      expect(reflectedHeader, isNull);
+
+      // Verify that auth is required by this endpoint method.
+      tokenInspectionCompleter = Completer();
+      int? statusCode;
+      try {
+        reflectedHeader =
+            await client.reflection.caller.callServerEndpoint<List<String>?>(
+          'reflection',
+          'reflectHttpHeader',
+          {
+            'headerName': 'authorization',
+          },
+        );
+      } catch (e) {
+        if (e is ServerpodClientException) {
+          statusCode = e.statusCode;
+        }
+      }
+      expect(statusCode, equals(401));
```

Comment:
Suggestion: Utalize`await expectLater` here to await the future with the correct value directly.

This can be used to catch any throws as well using the `throwsA` matcher.